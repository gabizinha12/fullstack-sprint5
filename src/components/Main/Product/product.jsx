import React from "react";
import {
  ListItem,
  ItemFigure,
  ItemLink,
  ItemImage,
  ItemFigcaption,
  ItemPrice,
} from "./styles";
function Product({ description, image, price, name }) {
  // const img = require(`../../${image}`);
  return (
    <ListItem>
      <ItemFigure>
        <ItemLink>
          <ItemImage src={image} alt="Produto" />
        </ItemLink>
        <ItemFigcaption>{name}</ItemFigcaption>
        <p className="card__description">{description}</p>
        <ItemPrice>
          <span>R$ {price}</span>
        </ItemPrice>
      </ItemFigure>
    </ListItem>
  );
}

export default Product;
