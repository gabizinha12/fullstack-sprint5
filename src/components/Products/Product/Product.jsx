import React from "react";
import { Link } from "react-router-dom";
import {
  ListItem,
  ItemFigure,
  ItemLink,
  ItemImage,
  ItemFigcaption,
  ItemPrice,
} from "./styles";
function Product({ description, image, price, name, sku }) {
  // const img = require(`../../${image}`);
  return (
    <ListItem>
      <ItemFigure key={sku}>
        <ItemLink to={`/product/${sku}`}>
          <ItemImage src={image} alt="Produto" />
        </ItemLink>
        <ItemFigcaption>{name}</ItemFigcaption>
        <p className="card__description">{description}</p>
        <ItemPrice>
          <span>R$ {price}</span>
        </ItemPrice>
      </ItemFigure>
    </ListItem>
  );
}

export default Product;
