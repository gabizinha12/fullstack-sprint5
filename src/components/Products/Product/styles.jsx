import styled from "styled-components";

export const ListItem = styled.li`
  flex-basis: 24%;
  text-align: center;
  font-size: 0.625rem;
  line-height: 1.5;
  text-transform: uppercase;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 1px 1px 10px 1px;
  cursor: pointer;
  background-color: white;
  &:hover {
    box-shadow: -3px 10px 10px -10px #c0c0c0;
  }
`;

export const ItemFigure = styled.figure``;

export const ItemLink = styled.a``;

export const ItemImage = styled.img`
  width: 100%;
`;

export const ItemFigcaption = styled.figcaption`
  text-transform: uppercase;
  text-align: center;
  margin-top: 8px;
  margin-bottom: 10px;
  line-height: 1.3;
  font-weight: 400;
`;

export const ItemPrice = styled.figcaption`
  padding-bottom: 8px;
  span {
    text-transform: uppercase;
    text-align: center;
    font-weight: 700;
  }
`;
