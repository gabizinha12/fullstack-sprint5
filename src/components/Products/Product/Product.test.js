import { render, screen } from "@testing-library/react";
import Product from "./Product";
import { BrowserRouter } from "react-router-dom";

describe("Teste de snapshot produto", () => {
  it("should render product", () => {
    const { container } = render(
      <BrowserRouter>
        <Product
          description="teste"
          image="aaa"
          price="10,99"
          name="abc"
          sku="10"
        />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
