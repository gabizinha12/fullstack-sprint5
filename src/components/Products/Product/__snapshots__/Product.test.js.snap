// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot produto should render product 1`] = `
<li
  class="sc-bdnxRM egDJlx"
>
  <figure
    class="sc-gtsrHT"
  >
    <a
      href="/product-detail/10"
    >
      <img
        alt="Produto"
        class="sc-hKFxyN gcWuxB"
        src="aaa"
      />
      <figcaption
        class="sc-eCApnc dQrqnr"
      >
        abc
      </figcaption>
      <p
        class="card__description"
      >
        teste
      </p>
      <figcaption
        class="sc-jSFjdj bEIScR"
      >
        <span>
          R$ 
          10,99
        </span>
      </figcaption>
    </a>
  </figure>
</li>
`;
